lib/site_setting_extension.rb

```diff
@@ -442,12 +442,17 @@ def set(name, value, options = nil)
     end
   end
 
-  def set_and_log(name, value, user = Discourse.system_user)
+  def set_and_log(name, value, user = Discourse.system_user, detailed_message = nil)
     if has_setting?(name)
       prev_value = public_send(name)
       set(name, value)
       value = prev_value = "[FILTERED]" if secret_settings.include?(name.to_sym)
-      StaffActionLogger.new(user).log_site_setting_change(name, prev_value, value)
+      StaffActionLogger.new(user).log_site_setting_change(
+        name,
+        prev_value,
+        value,
+        detailed_message.present? ? { details: detailed_message } : {},
```

Comment:
You can replace this by: `{ details: detailed_message }.compact_blank` :slightly_smiling_face:

Reply:
Ooh nice, thanks!